### ISSUE TITLE
LPE fails on ACCEPT

### ISSUE THREAD

**Issue by pjljvandelaar** [TorXakis/TorXakis#853]
When making lpe transformation of
```
PROCDEF produce [ Out         :: Int
                ] ( id :: Int) ::=
    (       Out ! id
        >-> EXIT ! (id+1)
    )
    >>> ACCEPT ? nextId :: Int IN
        produce [Out] (nextId)
    NI
ENDDEF

CHANDEF Chans ::= Out   :: Int
ENDDEF

MODELDEF Model ::=
    CHAN IN
    CHAN OUT    Out
    
    BEHAVIOUR
        produce [Out] (0)
ENDDEF
```
one gets
```
PROCDEF LPE_proxyProcess [ Out :: Int ] ( pc$proxyProcess :: Int; produce$pre1$Out$id :: Int; produce$pre1$Out$pc$produce$pre1$lhs :: Int; produce$pre1$Out$produce$pre1$lhs$Out$id :: Int; produce$pre1$Out$produce$pre1$lhs$gnf1$Out$produce$pre1$lhs$gnf1$id :: Int; produce$Out$id :: Int )
::=
( HIDE [ Hidden$Channel :: Int ] IN
{ Hidden$Channel ? EXIT$1 } [[ ( IF ( 1 == pc$proxyProcess ) THEN ( IF ( 1 == produce$pre1$Out$pc$produce$pre1$lhs ) THEN ( EXIT$1 == ( 1 + produce$pre1$Out$produce$pre1$lhs$gnf1$Out$produce$pre1$lhs$gnf1$id ) ) ELSE False FI ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 2, produce$pre1$Out$id, produce$pre1$Out$pc$produce$pre1$lhs, produce$pre1$Out$produce$pre1$lhs$Out$id, produce$pre1$Out$produce$pre1$lhs$gnf1$Out$produce$pre1$lhs$gnf1$id, nextId, EXIT$1 )
)
NI )
##
( { Out ? Out$1 :: Int } [[ ( IF ( 0 == pc$proxyProcess ) THEN ( 0 == Out$1 ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 1, 0, 1, 0, 0, produce$Out$id )
) )
##
( { Out ? Out$1 :: Int } [[ ( IF ( 1 == pc$proxyProcess ) THEN ( IF ( 0 == produce$pre1$Out$pc$produce$pre1$lhs ) THEN ( Out$1 == produce$pre1$Out$produce$pre1$lhs$Out$id ) ELSE False FI ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 1, produce$pre1$Out$id, 1, produce$pre1$Out$produce$pre1$lhs$Out$id, produce$pre1$Out$produce$pre1$lhs$Out$id, produce$Out$id )
) )
##
( { Out ? Out$1 :: Int } [[ ( IF ( 2 == pc$proxyProcess ) THEN ( Out$1 == produce$Out$id ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 1, produce$Out$id, 1, produce$Out$id, produce$Out$id, produce$Out$id )
) )
ENDDEF
```
The first alternative contains `nextId` yet this is not defined.
@CarstenRuetz Should it be replaced by `EXIT$1`?

**Comment by pjljvandelaar** [comment]
Note without a direct process instantiation everything is fine.
For example, lpe applied on
```
PROCDEF produce [ Out         :: Int
                ] ( id :: Int) ::=
    (       Out ! id
        >-> EXIT ! (id+1)
    )
    >>> ACCEPT ? nextId :: Int IN
            Out ! nextId
        >-> produce [Out] (nextId)
    NI
ENDDEF

CHANDEF Chans ::= Out   :: Int
ENDDEF

MODELDEF Model ::=
    CHAN IN
    CHAN OUT    Out
    
    BEHAVIOUR
        produce [Out] (0)
ENDDEF
```
results in a correct transformation
```
PROCDEF LPE_proxyProcess [ Out :: Int ] ( pc$proxyProcess :: Int; produce$pre1$Out$id :: Int; produce$pre1$Out$pc$produce$pre1$lhs :: Int; produce$pre1$Out$produce$pre1$lhs$Out$id :: Int; produce$pre1$Out$produce$pre1$lhs$gnf1$Out$produce$pre1$lhs$gnf1$id :: Int; produce$pre1$rhs$Out$id :: Int; produce$pre1$rhs$Out$nextId :: Int; produce$Out$id :: Int )
::=
( HIDE [ Hidden$Channel :: Int ] IN
{ Hidden$Channel ? EXIT$1 } [[ ( IF ( 1 == pc$proxyProcess ) THEN ( IF ( 1 == produce$pre1$Out$pc$produce$pre1$lhs ) THEN ( EXIT$1 == ( 1 + produce$pre1$Out$produce$pre1$lhs$gnf1$Out$produce$pre1$lhs$gnf1$id ) ) ELSE False FI ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 2, produce$pre1$Out$id, produce$pre1$Out$pc$produce$pre1$lhs, produce$pre1$Out$produce$pre1$lhs$Out$id, produce$pre1$Out$produce$pre1$lhs$gnf1$Out$produce$pre1$lhs$gnf1$id, produce$pre1$Out$id, EXIT$1, produce$Out$id )
)
NI )
##
( { Out ? Out$1 :: Int } [[ ( IF ( 0 == pc$proxyProcess ) THEN ( 0 == Out$1 ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 1, 0, 1, 0, 0, produce$pre1$rhs$Out$id, produce$pre1$rhs$Out$nextId, produce$Out$id )
) )
##
( { Out ? Out$1 :: Int } [[ ( IF ( 1 == pc$proxyProcess ) THEN ( IF ( 0 == produce$pre1$Out$pc$produce$pre1$lhs ) THEN ( Out$1 == produce$pre1$Out$produce$pre1$lhs$Out$id ) ELSE False FI ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 1, produce$pre1$Out$id, 1, produce$pre1$Out$produce$pre1$lhs$Out$id, produce$pre1$Out$produce$pre1$lhs$Out$id, produce$pre1$rhs$Out$id, produce$pre1$rhs$Out$nextId, produce$Out$id )
) )
##
( { Out ? Out$1 :: Int } [[ ( IF ( 2 == pc$proxyProcess ) THEN ( Out$1 == produce$pre1$rhs$Out$nextId ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 3, produce$pre1$Out$id, produce$pre1$Out$pc$produce$pre1$lhs, produce$pre1$Out$produce$pre1$lhs$Out$id, produce$pre1$Out$produce$pre1$lhs$gnf1$Out$produce$pre1$lhs$gnf1$id, produce$pre1$rhs$Out$id, produce$pre1$rhs$Out$nextId, produce$pre1$rhs$Out$nextId )
) )
##
( { Out ? Out$1 :: Int } [[ ( IF ( 3 == pc$proxyProcess ) THEN ( Out$1 == produce$Out$id ) ELSE False FI ) ]]
>-> ( LPE_proxyProcess [Out] ( 1, produce$Out$id, 1, produce$Out$id, produce$Out$id, produce$pre1$rhs$Out$id, produce$pre1$rhs$Out$nextId, produce$Out$id )
) )
ENDDEF
```

**Comment by pjljvandelaar** [comment]
Note out commenting the case of a ProcInst works for this issue, yet multiple tests result in errors and failures. So does not seem to be the solution.
@CarstenRuetz We need to substitute the expressions of the parameters. Any idea? The parameter changes are done later it seems.....

**Comment by pjljvandelaar** [comment]
Found another bug in LPE
```
procIdR = procIdInst {  ProcId.name = name',
                                     ProcId.unid = unidR,
                                     -- concatenate the original params with the varIds passed via ACCEPT to extend their scope into the RHS of ENABLE
                                     ProcId.procvars = varsort <$> paramsDef ++ paramsAccept} 
-- create ProcInst, translate params to VExprs
paramsDef' = map cstrVar paramsDef
procInstR' = procInst procIdR chansDef paramsDef'
```
`procIdR` has more parameters than `paramsDef` provide!

**Comment by pjljvandelaar** [comment]
solved by #853
